### PR TITLE
add AutoDJ xfader recenter option (default off)

### LINKED ISSUE #11571: Re-add Crossfader re-center when disabling auto DJ
### Feature Description

I would like to request for the crossfader auto re-set to be re-instated to the next update of Mixxx.

Thank You.

### PR DESCRIPTION
**Pull request by ronso0**
adopting #13156 but cleaning up the "buffer" settings before
(that dlg wrote some "...Buff" settings to config for storing temp pending settings...)

Fixes #11571

### TODO:
- [x] add a note how re-centering may affect main volume (see #10683 and #4714)

### CODE REVIEW

**Comment by Eve00000** [comment]
Hi Ronso,
I was almost ready to upload my adapted version, but you already did the changes. You cleaned up the whole routine. Thank you for the work and congratulations it surely will be better then mine .
I will learn by comparing your version with what I did.

If it's possible I just want to add the lines in the ui with some explanation (tooltip),
I don't know how to add that in the pr so I'll write it here.

<property name="toolTip">
<string>When checked, the Crossfader will be set to the default center position after disabling the Auto DJ.
This can be handy when you're working with an external mixer and don't need to see the Mixxx-mixer and crossfader 
(mixer and/or crossfader disabled in the skin-options).

Please mind that the crossfader adds some volume to the pointed playing deck in the main output, 
so centering the crossfader will cause a volumedrop in the main output.
</string>
</property>

greetings

**Review by Swiftb0y — COMMENTED** [review]
couple comments, looks pretty good so far. I can't really review the ui file though

File: src/preferences/dialog/dlgprefautodj.cpp [hunk 1]
```diff
@@ -57,6 +57,10 @@ void DlgPrefAutoDJ::slotUpdate() {
                     ConfigKey("[Auto DJ]", "EnableRandomQueue"))
                     ? Qt::Checked
                     : Qt::Unchecked);
+
+    // Re-center the crossfader instantly when AutoDJ is disabled
+    CenterXfaderCheckBox->setChecked(m_pConfig->getValue(
+            ConfigKey("[Auto DJ]", "center_xfader_when_disabling"), false));
```
[the remark this answers is not in the document]
Comment (ronso0) [reply src/preferences/dialog/dlgprefautodj.cpp:63]: Jep, moving all repeatedly used pref ConfigKeys (preferences, and respective engine units) to other files.
I have some changes stashed already, but I prefer to do that in a separate PR.

File: src/library/autodj/autodjprocessor.cpp [hunk 2]
```diff
@@ -578,6 +578,11 @@ AutoDJProcessor::AutoDJError AutoDJProcessor::toggleAutoDJ(bool enable) {
         for (const auto& pDeck : std::as_const(m_decks)) {
             pDeck->disconnect(this);
         }
+        // if (m_pConfig->getValue<bool>(ConfigKey(kAutoDJGroup,
+        if (m_pConfig->getValue<bool>(ConfigKey(kConfigKey,
+                    QStringLiteral("center_xfader_when_disabling")))) {
```
[the remark this answers is not in the document]
Comment (ronso0) [reply src/library/autodj/autodjprocessor.cpp:583]: whoopsy

File: src/library/autodj/autodjprocessor.cpp [hunk 3]
```diff
@@ -578,6 +578,11 @@ AutoDJProcessor::AutoDJError AutoDJProcessor::toggleAutoDJ(bool enable) {
...
+            m_pCOCrossfader->set(0);
```
[the remark this answers is not in the document]
Comment (Swiftb0y) [reply src/library/autodj/autodjprocessor.cpp:583]: fair. I just thought the `_set_default` was more expressive.

**Review by Swiftb0y — APPROVED** [review]
LGTM. feel free to squash and force push

**Comment by Swiftb0y** [comment]
Sorry, forgot about this.

**Pull request merged**